src/main/java/co/alphavantage/common/lib/Function.java

```diff
@@ -1,32 +1,34 @@
-package co.alphavantage.input.timeseries;
-
-import co.alphavantage.input.ApiParameter;
-
-/**
- * Function parameter for the time series api call.
- */
-public enum Function implements ApiParameter {
-  INTRADAY("TIME_SERIES_INTRADAY"),
-  DAILY("TIME_SERIES_DAILY"),
-  DAILY_ADJUSTED("TIME_SERIES_DAILY_ADJUSTED"),
-  WEEKLY("TIME_SERIES_WEEKLY"),
-  WEEKLY_ADJUSTED("TIME_SERIES_WEEKLY_ADJUSTED"),
-  MONTHLY("TIME_SERIES_MONTHLY"),
-  MONTHLY_ADJUSTED("TIME_SERIES_MONTHLY_ADJUSTED");
-
-  private final String urlParameter;
-
-  Function(String urlParameter) {
-    this.urlParameter = urlParameter;
-  }
-
-  @Override
-  public String getKey() {
-    return "function";
-  }
-
-  @Override
-  public String getValue() {
-    return this.urlParameter;
-  }
-}
+package co.alphavantage.common.lib;
+
+import co.alphavantage.input.ApiParameter;
+
+public enum Function implements ApiParameter {

```

Comment:
The problem with having only one Function Enum is that now all api:s (TimeSeries, CurrencyExchange, etc..) share the same Function Enum, and there is no way to ensure that only the right Function enums gets passed to the right api. The reason for having separate Function enums was to ensure that only correct function can be passed.

Reply:
This Function primes abstraction. Right now it is pretty hard coupled. My 2 cents.

Reply:
Agree to have only the functions inside a file. I think that @patriques82, you are saying that to avoid the developer putting wrong functionality but in this case the developer must handle the functionality (not the user). For that reason it has no sense to create multiple files.

Reply:
That is a good point. Lets keep the change.